### PR TITLE
async100 now ignores trio.open_nursery and anyio.create_task_group

### PR DESCRIPTION
**Pull request by jakkdl**
I was cleaning up old branches in my fork, and it... seems like I never opened a PR for this? Wrote it several months ago, so decided to just rebase and clean it up

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -4,6 +4,10 @@ Changelog
 
 `CalVer, YY.month.patch <https://calver.org/>`_
 
+24.11.1
+======
+- :ref:`ASYNC100 <async100>` now ignores :func:`trio.open_nursery` and :func:`anyio.create_task_group` as cancellation sources.
```
Comment (Zac-HD) [inline docs/changelog.rst:9]:
```suggestion
- :ref:`ASYNC100 <async100>` now ignores :func:`trio.open_nursery` and :func:`anyio.create_task_group` as cancellation sources,
  because they are schedule points but not cancellation points.
```

This is a subtle enough concept that we should not assume users are familiar with it - I'd also call out this distinction in the docs for `ASYNC100`, and create glossary entries for both which are linked from "checkpoint" and link in turn to Trio docs (eg for `cancel_shielded_checkpoint()` and `checkpoint_if_cancelled()`)

**Review by Zac-HD — APPROVED** [review]
LGTM with some more documentation as below - thanks!

File: docs/changelog.rst (hunk 1 above)
Comment (jakkdl) [reply docs/changelog.rst:9]: good call, feel free to reword my attempt :)

File: docs/changelog.rst [hunk 2]
```diff
@@ -23,6 +27,7 @@ Changelog
 24.9.3
 ======
 - :ref:`ASYNC102 <async102>` and :ref:`ASYNC120 <async120>`:
+
   - handles nested cancel scopes
```
Comment (jakkdl) [inline docs/changelog.rst:31]: unrelated fix: this was not getting parsed as a nested bullet list w/o an empty line

**Review by Zac-HD — APPROVED** [review]
Added some tiny tweaks to the docs for cross-references.  Thanks again @jakkdl!
